src/java.base/windows/classes/sun/nio/fs/WindowsFileSystemProvider.java

```diff
@@ -429,40 +430,53 @@ public boolean isSameFile(Path obj1, Path obj2) throws IOException {
             return false;
         WindowsPath file2 = (WindowsPath)obj2;
 
-        // open both files and see if they are the same
-        long h1 = 0L;
-        try {
-            h1 = file1.openForReadAttributeAccess(true);
-        } catch (WindowsException x) {
-            x.rethrowAsIOException(file1);
-        }
-        try {
-            WindowsFileAttributes attrs1 = null;
+        // check existence
+        boolean exists1 = exists(obj1);
+        boolean exists2 = exists(obj2);
```

Comment:
exists does the openForReadAttributeAccess+readAttributes+CloseHandle so it looks like the change means it will do it twice for each file.

Reply:
Right, the Windows implementation is not good and needs to be rethought.